### DISCUSSION TITLE
Can we use `middleware.URLFormat` inside mount?

### DISCUSSION THREAD

**Discussion by josejibin** [go-chi/chi#717] [Q&A]
Maybe a code sample give more context.

```
package main

import (
	"flag"
	"net/http"

	"github.com/go-chi/chi/v5"
	"github.com/go-chi/chi/v5/middleware"
)

func main() {
	flag.Parse()

	r := chi.NewRouter()
	r.Use(middleware.Logger)

	r.Get("/ping", func(w http.ResponseWriter, r *http.Request) {
		w.Write([]byte("pong"))
	})

	r.Mount("/admin", adminRouter())

	http.ListenAndServe(":3333", r)
}

// A completely separate router for administrator routes
func adminRouter() chi.Router {
	r := chi.NewRouter()
	r.Use(middleware.URLFormat)
	r.Get("/test", func(w http.ResponseWriter, r *http.Request) {
		w.Write([]byte("route found!!"))
	})
	return r
}
```
try `curl -X GET 'http://localhost:3333/admin/test.json'`

Expected output:
```
route found!!
```

Actual output:
```
404 page not found
```
`curl -X GET 'http://localhost:3333/admin/test'` is working as expected, though. So `middleware.URLFormat` doesn't work with `mount`?

**Comment by tamish-max** [comment]
`URLFormat` has the same caveat as `StripSlashes`: it needs to run on the router that is doing the path matching. Inside a mounted subrouter, `/test.json` still fails to match `/test` before the handler chain is built, so the middleware never gets a chance to trim `.json`.

Try applying it on the parent router **before** the mount:

```go
r := chi.NewRouter()
r.Use(middleware.URLFormat)
r.Mount("/admin", adminRouter())
```

or match the suffix explicitly (`/test.{format}`). The key is that `URLFormat` must run early enough to rewrite the route path before matching.
